### PR TITLE
Add support for cip 66 transactions

### PR DESCRIPTION
**Pull request by aaronmgdr**
## what

* Adding support for the [CIP66 Transaction Type](https://github.com/celo-org/celo-proposals/blob/master/CIPs/cip-0066.md) This upcoming transaction type will be supported on CEL2. 

* Defaults to using cip 66 if sending feeCurrency Transaction on a chain that supports it.#22 

* add support for automatically calculating fee fields needed for cip66

* add isCel2 helper which can be used by anyone to determine if chain is an OP based Layer 2 or still old Layer 1

## bump type

minor; adds new features

## other

* separate transaction tests that dont use the network / need constant results into new unit tests file. 

* remove ./tests from the `test`command. this added nothing and prevented adhoc filters being applied to which tests were ran like `yarn test CeloWallet`

### CODE REVIEW

File: src/lib/CeloWallet.ts [hunk 1]
```diff
@@ -23,6 +25,16 @@ const forwardErrors = [
 ] as ErrorCode[];
 
 export default class CeloWallet extends Wallet {
+
+  async isCel2(){
```
Comment (nicolasbrugneaux) [inline src/lib/CeloWallet.ts:29]: this rings a bell hehe

File: tests/transactions.unit.test.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { describe, expect, it } from "@jest/globals";
```
Comment (nicolasbrugneaux) [inline tests/transactions.unit.test.ts:1]: I love the new file, any reason for not porting over the other tx tests? 

Not super important though.

Comment (aaronmgdr) [reply tests/transactions.unit.test.ts:1]: the other tests in transactions.test.ts actually send transaction on the blockchain and as such are not unit tests.

File: package.json [hunk 3]
```diff
@@ -17,7 +17,7 @@
     "build": "yarn build:main && yarn build:module",
     "build:main": "tsc -p tsconfig.json",
     "build:module": "tsc -p tsconfig.module.json",
-    "test": "jest ./tests"
+    "test": "jest"
```
Comment (aaronmgdr) [inline package.json:20]: not needed and means that calling yarn test <filter> does not work

File: src/lib/transactions.ts [hunk 4]
```diff
@@ -132,15 +143,39 @@ export function getTxType(tx: CeloTransaction) {
...
+      raw = [
+        toBeHex(tx.chainId!),
+        toBeHex(tx.nonce!),
+        tx.maxPriorityFeePerGas ? toBeHex(tx.maxPriorityFeePerGas) : "0x",
+        tx.maxFeePerGas ? toBeHex(tx.maxFeePerGas) : "0x",
+        tx.gasLimit ? toBeHex(tx.gasLimit) : "0x",
+        tx.to || "0x",
+        tx.value ? toBeHex(tx.value) : "0x",
+        tx.data || "0x",
+        // @ts-expect-error
```
Comment (jmrossy) [inline src/lib/transactions.ts:173]: I've had problems with ts-expect-error silencing the error here in this lib's build but not silencing it properly in lib consumers (unless they have skipLibCheck enabled). Recommend ts-ignore. WDYT?

File: src/lib/CoreContract.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      }
+    ],
+    "payable": false,
+    "stateMutability": "view",
+    "type": "function"
+  }]
+
+
+  
+  export async function getConversionRateFromCeloToToken(tokenAddress: string, {wallet}: {wallet: CeloWallet}): Promise<[bigint, bigint]> {
```
[the remark this answers is not in the document]
Comment (aaronmgdr) [reply src/lib/CoreContract.ts:59]: no longer a good reason

File: src/lib/transactions.ts (hunk 4 above)
Comment (aaronmgdr) [reply src/lib/transactions.ts:173]: yeah now im questioning why it needs this at all and realizing that access list transactions wouldnt work. so ill fix that.

File: tests/CeloWallet.test.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+              expect(filled.maxFeeInFeeCurrency).toBeUndefined()
+              expect(filled.feeCurrency).toBeUndefined()
+
+            })
+        })
+        describe('when feeCurrency on cel2', () => {
+            test('populates maxFeePerGas and maxPriorityFeePerGas and maxFeeInFeeCurrency', async () => {
+              const wallet = getSigner()
+
+
```
[the remark this answers is not in the document]
Comment (aaronmgdr) [reply tests/CeloWallet.test.ts]: ran prettier on the changed files.

**Review by jmrossy — APPROVED** [review]

**Pull request merged**